snapper/AppUtil.cc

```diff
@@ -182,19 +191,23 @@ namespace snapper
 
 	found = false;
 
-	struct mntent* m;
-	while ((m = getmntent(f)))
+	struct mntent m;
+	// each mnt table element is limited to PAGE_SIZE top in Linux
```

Comment:
Hello,

well I have to admit it's not (at least I can't find it directly mentioned anywhere). So I'm sorry for the misleading comment (I've mistaken /proc for /sys attributes in my mind). But I think it's good  compromise anyway. The mnt ent elements are as follows:

a) device is path, or UUID (PATH_MAX?)
b) mnt dir is path (PATH_MAX?)
c) fs type ???
d) mnt options ???. well, there're filesystems accepting an option with path as an argument so....PATH_MAX +more?

I've seen various buffer sizes in use with getmntent_r: 2*PATH_MAX (wrong due to above)
single PAGE_SIZE - wrong (wrong due to above). This is in glibc getmntent implementation btw (which calls getmntent_r internally).

But I'm open to suggestions